docs/plugin_developers_guide/11_WSL2.rst

```diff
@@ -0,0 +1,328 @@
+Since you are launching Visual Studio Code from your Linux distribution, Code should prompt you to install two Extensions:
+
+* `Remote-WSL <https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl>`_
+* `C/C++ <https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools>`_
+
+If you are not prompted, follow these links to install the extensions and restart Visual Studio Code.
+If the extensions are installed and working, you should see a green label in the lower left-hand corner of the Visual Studio Code window
+with the text *WSL:* followed by the name of your Linux distribution.
+
+Configuring your Workspace
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Visual Studio Code refers to your directory of source code files as the *Workspace.*
+In order to edit, build and debug your code, you must create 3 Json files in a Workspace subdirectory called *.vscode*:
+
+* **c_cpp_properties.json**: compiler path, IntelliSense settings, and include file paths,
+* **tasks.json**: build instructions,
+* **launch.json**: debugger settings.
+
+You can create these files manually or use Visual Studio Code's configuration wizards.
+These subsections describe creation and required contents of each of these three files.
+
+Code Editor Configuration: c_cpp_properties.json
+#####################################################
+
+Open the Command Palette using the key sequence *Ctrl+Shift+P*.
+Choose the command *C/C++: Edit Configurations (JSON)*.
+This will create the *.vscode* subdirectory (if it doesn't already exist) and the *c_cpp_properties.json* file.
+This Json file will be opened for editing.
+You will see a new array called *configurations* with a single configuration object defined.
+This configuration will have a string array called *includePath*.
+Add the paths to your own include files, and those required by the Fledge API to the *includePath* array.
+You can use Linux environment variables in your paths.
+For example:
+
+.. code-block:: json
+
+   "${FLEDGE_ROOT}/C/common/include"
+
+You can find the list of include files by running your *make* command:
+
+.. code-block:: bash
+
+   make --just-print
+   
+which will list all commands defined by *make* without executing them.
+You will see the include file list in every instance of the *gcc* compiler command.
+
+Build Configuration: tasks.json
+###############################
+
+From the Visual Studio Code main menu, choose *Terminal -> Configure Default Build Task*.
+A dropdown will display of available tasks for C++ projects.
+Choose *g++ build active file*.
+This will create the *.vscode* subdirectory (if it doesn't already exist) and the *tasks.json* file.
+Open the Json file for editing.
+
+Building the project will be done using the *make* file rather than the *gcc* compiler.
+To make this change, edit the *command* and *args* entries as follows:
+
+.. code-block:: json
+
+   "command": "make",
+   "args": [
+      "-C",
+      "${workspaceFolder}/build"
+   ],
```

Comment:
This is not a valid JSON code snippet. We may use `console` or any other directive